### ISSUE TITLE
'ID must be unique' error message

### ISSUE THREAD

**Issue by madhurrya** [sartography/spiff-arena#2052]
In https://mysql.spiffcrm.com/ when I save an existing model it's showing this 'ID must be unique' error message in properties. But the model get saved successfully.
As discussed on slack those model ids actually doesn't exist in the system. It seems this is not a valid message. Seems to be due to a recent change. If I have not expanded the Process general section, I wouldn't even notice this message as the model get saved successfully.

![Image](https://github.com/user-attachments/assets/0ed0bb99-aa84-404d-992c-a76c41e670e4)

These are some models you can check this
https://mysql.spiffcrm.com/editor/process-models/misc:qa:madhu-testing:temp-model1/files/temp-model1.bpmn
https://mysql.spiffcrm.com/editor/process-models/misc:qa:call-activity-with-non-executable-model/files/call-activity-with-non-executable-model.bpmn

If the ID actually exists in another model, it normally shows this error message at the top like this. But for the ones I mentioned above there is no error message at the top. I see it only if I expand the process properties.

![Image](https://github.com/user-attachments/assets/7a0dac78-1ce4-40d6-a7ab-edb403c2b9f7)

**Comment by jasquat** [comment]
The `ID must be unique.` error in the properties panel is not something we control; it's handled by bpmn-io libraries that we use. We did recently update some of those libraries though. It could be that the new versions are more sensitive.

This error also won't keep you from saving the model, however it will keep the xml from getting updated so even if you save, the supposedly invalid id should not be passed through.

One thing to keep in mind, bpmn-js-properties-panel requires ALL ids to be unique. So even if a script task has a matching id, it will show this error.

**Comment by calexh-sar** [comment]
> One thing to keep in mind, bpmn-js-properties-panel requires ALL ids to be unique. So even if a script task has a matching id, it will show this error.

@usama9500, in case this isn't documented somewhere.  Maybe should be both in our core documentation and FAQ?

**Comment by calexh-sar** [comment]
@madhurrya and/or @jasquat is this an BPMN.IO issue and we should move it to that repository, or is it a non-issue and making sure the unique id aspect is documented is all that is needed?

**Comment by jasquat** [comment]
@calexh-sar i think we should just document it unless we confirm that we are seeing false positives.

**Comment by madhurrya** [comment]
Noticed the same issue in spiffdemo also. I just created this new model and when trying to save it noticed the same error.
![image](https://github.com/user-attachments/assets/c3da1bbc-0126-4c3b-bb8f-2b8380b44257)

Model - https://spiffdemo.org/editor/process-models/playground:madhurya:madhu-test-form-1/files/madhu-test-form-1.bpmn

@jasquat is there anything we can do about this?

**Comment by jasquat** [comment]
@madhurrya I don't think there is anything we can do about this easily. It looks like the ID you have in the screenshot is currently the id of the Process so you did manage to save that ID. What were the steps you took to get this error?

**Comment by madhurrya** [comment]
@jasquat Yes, even if that error comes the model get saved successfully.
For the above model what I did was created a new model and changed the default manual task to user task and renamed it and changed the instructions and added a form and saved.
Followed the same steps and it happened here again. Please check this video. https://drive.google.com/file/d/1r4CTJccxhi3UIdS8IplLysifURrqBbXG/view?usp=sharing

But in this video it didn't happen for the above steps. But after saving it couple of times again then the error came.
https://drive.google.com/file/d/1GglEH8uXezfHqa8VPVjpHNcJDCAhEHte/view?usp=sharing

I feel this happens randomly and not specific to a model or the steps we follow.

**Comment by jasquat** [comment]
Oh odd. It seems to happen when you save the diagram while some element is selected, such as the End Event or User task. If no item is selected when saving, then it doesn't seem to happen. This could be due to the way we are refreshing the state or something but it is rather odd.

**Comment by madhurrya** [comment]
@jasquat yeah it seems that's what happens. I checked again now and it seems if I save while an element is selected it gives that error. But if I have not selected then no error.
